hi-ml/src/health_ml/runner.py

```diff
@@ -320,15 +324,42 @@ def run(project_root: Path) -> Tuple[LightningContainer, AzureRunInfo]:
     """
     The main entry point for training and testing models from the commandline. This chooses a model to train
     via a commandline argument, runs training or testing, and writes all required info to disk and logs.
+    When running in Azure, this method also redirects the stdout stream, so that all console output is visible both on
+    the console and stored in a file. The filename is timestamped and contains the DDP rank of the current process.
 
     :param project_root: The root folder that contains all of the source code that should be executed.
     :return: If submitting to AzureML, returns the model configuration that was used for training,
-    including commandline overrides applied (if any). For details on the arguments, see the constructor of Runner.
+        including commandline overrides applied (if any). For details on the arguments, see the constructor of Runner.
     """
     if is_global_rank_zero():
-        print(f"project root: {project_root}")
-    runner = Runner(project_root)
-    return runner.run()
+        print(f"Project root: {project_root}")
+    if is_running_in_azure_ml():
+        # Create a timestamped filename. This will also ensure that all restarts after low-priority preemption create
+        # a new log file, and we can fully trace back what happened in each rank in each restart.
```

Comment:
[Minor] May I suggest that we write a helper function for lines 337-348 : get_logging_filename()
it makes it easier to test and improves code readability

Reply:
done & tested